Dashboard/app/js/lib/components/devices/AssignmentsList.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+/* eslint-disable no-nested-ternary */
+const AssignmentsList = ({
+  assignments,
+  onEdit,
+  onDelete,
+  onSort,
+}) => (
+  <div>
+    <div className="deviceControls">
+      <a
+        className="btnOutline"
+        onClick={() => onEdit('new')}
+        onKeyDown={() => onEdit('new')}
+        style={{ float: 'right' }}
+      >
+        {Ember.String.loc('_create_new_assignment')}
+      </a>
+    </div>
+    <div
+      id="devicesListTable_length"
+      className="dataTables_length"
+    />
+    {!assignments.get('length') && (
+      <div style={{ marginTop: '40px' }}>No assignments</div>
+    )}
+    {assignments.get('length') > 0 && (
+      <table className="dataTable">
+        {/* TABLE HEADER */}
+        <thead>
+          <tr>
+            <th
+              className={FLOW.tableColumnControl.get('selected') == 'name' ? (FLOW.tableColumnControl.get('sortAscending') ? 'sorting_asc' : 'sorting_desc') : ''}
+            >
+              <a
+                onClick={() => onSort('name')}
+                onKeyDown={() => onSort('name')}
+              >
+                {Ember.String.loc('_name')}
+              </a>
+            </th>
+            <th
+              className={FLOW.tableColumnControl.get('selected') == 'startDate' ? (FLOW.tableColumnControl.get('sortAscending') ? 'sorting_asc' : 'sorting_desc') : ''}
+            >
+              <a
+                onClick={() => onSort('startDate')}
+                onKeyDown={() => onSort('startDate')}
+              >
+                {Ember.String.loc('_start_date')}
+              </a>
+            </th>
+            <th
+              className={FLOW.tableColumnControl.get('selected') == 'endDate' ? (FLOW.tableColumnControl.get('sortAscending') ? 'sorting_asc' : 'sorting_desc') : ''}
+            >
+              <a
+                onClick={() => onSort('endDate')}
+                onKeyDown={() => onSort('endDate')}
+              >
+                {Ember.String.loc('_end_date')}
+              </a>
+            </th>
+            <th className="noArrows">
+              <a>
+                {Ember.String.loc('_action')}
+              </a>
+            </th>
+          </tr>
+        </thead>
+        {/* TABLE BODY: MAIN CONTENT */}
+        <tbody>
+          {assignments.map(assignment => (
+            <tr key={assignment.get('keyId')}>
+              <td className="name">{assignment.get('name')}</td>
```

Comment:
I know this is probably annoying but is it not possible for assignments to be an array of plain objects so we can avoid having to say `assignment.get('name')` and instead say `assignment.name`? This is because it is a more standard format of data/props and we can more easily use this component later elsewhere